### PR TITLE
Synchronize security-jwt.adoc with smallrye-jwt

### LINKED ISSUE #13091: Document config property smallrye.jwt.verify.key.location
**Description**
As already noted in https://github.com/quarkusio/quarkus/issues/9791#issuecomment-713769020 the current version of the [JWT guide](https://quarkus.io/guides/security-jwt#additional-smallrye-jwt-configuration) does not document the `smallrye.jwt.verify.key.location` property which was added in `smallrye-jwt` 2.3.0 (see https://github.com/smallrye/smallrye-jwt/pull/305). The current Quarkus 1.9.1 is using `smallrye-jwt` 2.3.1.

**Implementation ideas**
Sync the guide with the relevant version of https://github.com/smallrye/smallrye-jwt/blob/master/doc/modules/ROOT/pages/configuration.adoc.

### PR DESCRIPTION
**Pull request by knutwannheden**
Copy over the relevant portion of https://github.com/smallrye/smallrye-jwt/blob/master/doc/modules/ROOT/pages/configuration.adoc to security-jwt.adoc.

Fixes #13091

### CODE REVIEW

**Review by sberyozkin — APPROVED** [review]

File: docs/src/main/asciidoc/security-jwt.adoc [hunk 1]
```diff
@@ -661,24 +661,25 @@ SmallRye JWT provides more properties which can be used to customize the token p
...
 |smallrye.jwt.token.kid|none|Key identifier. If it is set then the verification JWK key as well every JWT token must have a matching `kid` header.
 |smallrye.jwt.time-to-live|none|The maximum number of seconds that a JWT may be issued for use. Effectively, the difference between the expiration date of the JWT and the issued at date must not exceed this value.
-|smallrye.jwt.require.named-principal|`false`|If an application relies on `java.security.Principal` returning a name then a token must have a `upn` or `preferred_username` or `sub` claim set. Setting this property will result in SmallRye JWT throwing an exception if none of these claims is available for the application code to reliably deal with a non-null `Principal` name.
+|smallrye.jwt.require.named-principal|`false`|If an application relies on `java.security.Principal` returning a name then a token must have a `upn` or `preferred_username` or `sub` claim set. Setting this property will result in Smallrye JWT throwing an exception if none of these claims is available for the application code to reliably deal with a non-null `Principal` name.
```
Comment (gsmet) [inline docs/src/main/asciidoc/security-jwt.adoc:675]: Well, this is not right so it should probably be fixed in the source. It should be `SmallRye`.

Comment (knutwannheden) [reply docs/src/main/asciidoc/security-jwt.adoc:675]: Yes, I noticed... Let me revert that and fix it in `smallrye-jwt`. I literally copy-and-pasted the corresponding lines from there.

Comment (knutwannheden) [reply docs/src/main/asciidoc/security-jwt.adoc]: Note: The copy-and-paste also explains the other minor differences (added backticks).

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**